Show constituency polygons on the map

This might be a tricky one. I think the client will need to request the GeoJSON file from the server and then probably the Mapbox GL JS package we're using has some built in functionality to render GeoJSON as a layer.

Could also add a convenience function to the data loading script to automatically fetch and save the GeoJSON file for us instead of manually retrieving it like currently specified.

I seem to have a problem with mongo DB here. Will try to fix it first.

`MongoServerSelectionError: connect ECONNREFUSED 127.0.0.1:27017
    at Timeout._onTimeout (/Users/jonlee/Documents/GitHub/f28cd-project/node_modules/mongodb/lib/core/sdam/topology.js:438:30)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7) {
  reason: TopologyDescription {
    type: 'Single',
    setName: null,
    maxSetVersion: null,
    maxElectionId: null,
    servers: Map(1) {
      'localhost:27017' => ServerDescription {
        address: 'localhost:27017',
        error: Error: connect ECONNREFUSED 127.0.0.1:27017
            at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
          name: 'MongoNetworkError'
        },
        roundTripTime: -1,
        lastUpdateTime: 128672376,
        lastWriteDate: null,
        opTime: null,
        type: 'Unknown',
        topologyVersion: undefined,
        minWireVersion: 0,
        maxWireVersion: 0,
        hosts: [],
        passives: [],
        arbiters: [],
        tags: []
      }
    },
    stale: false,
    compatible: true,
    compatibilityError: null,
    logicalSessionTimeoutMinutes: null,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    commonWireVersion: null
  }
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! f28cd-project@0.0.1 start:server: `node server/server.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the f28cd-project@0.0.1 start:server script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jonlee/.npm/_logs/2021-03-18T13_53_39_130Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! f28cd-project@0.0.1 start: `npm run build && npm run start:server`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the f28cd-project@0.0.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jonlee/.npm/_logs/2021-03-18T13_53_39_172Z-debug.log`

--- a/public/src/map.js
+++ b/public/src/map.js
@@ -40,7 +40,8 @@ export async function initMap(apiKey) {
     // polygon GeoJson
     map.addSource('polygons', {
       type: 'geojson',
-      data: '..public/assets/constituencies.geojson'
+      data: 'https://opendata.arcgis.com/datasets/937997590f724a398ccc0100dbd9feee_0.geojson'
+      // data: '..public/assets/constituencies.geojson'
     })
   
     // add polygon

--- a/public/src/map.js
+++ b/public/src/map.js
@@ -36,6 +36,24 @@ export async function initMap(apiKey) {
   // After the map style has loaded on the page,
   // add a source layer and default styling for a single point
   map.on('load', function () {
+    
+    map.addSource('polygons', {
+      type: 'geojson',
+      data: 'https://opendata.arcgis.com/datasets/937997590f724a398ccc0100dbd9feee_0.geojson'
+    })
+  
+    map.addLayer({
+      'id': 'polygons',
+      'type': 'fill',
+      'source': 'polygons',
+      'layout': {},
+      'paint': {
+      'fill-color': '#088',
+      'fill-opacity': 0.8,
+      'fill-outline-color': 'rgba(0,0,0,0)'
+      }
+    });
+    
     map.addSource('single-point', {
       type: 'geojson',
       data: {

--- a/public/src/map.js
+++ b/public/src/map.js
@@ -66,7 +66,7 @@ export async function initMap(apiKey) {
       'layout': {},
       'paint': {
         'line-color': '#0d324d',
-        'line-width': 4
+        'line-width': 2
       }
     });
     

--- a/public/src/map.js
+++ b/public/src/map.js
@@ -44,7 +44,7 @@ export async function initMap(apiKey) {
       // data: '..public/assets/constituencies.geojson'
     })
   
-    // add polygon
+    // add polygon fill
     map.addLayer({
       'id': 'polygons-fill',
       'type': 'fill',
@@ -69,7 +69,26 @@ export async function initMap(apiKey) {
         'line-width': 2
       }
     });
+
+    // adding a symbol layer
+    map.addLayer({
+      'id': 'polygons-label',
+      'type': 'symbol',
+      'source': 'polygons',
+      'layout': {
+        'icon-image': 'custom-marker',
+        // get the title name from the source's "title" property
+        'text-field': ['get', 'pcon19nm'],
+        'text-font': [
+          "Open Sans Regular",
+          "Arial Unicode MS Regular"
+        ],
+        'text-offset': [0, 1.25],
+        'text-anchor': 'top'
+      }
+    });
     
+    // Add source fo search pin
     map.addSource('single-point', {
       type: 'geojson',
       data: {
@@ -78,6 +97,7 @@ export async function initMap(apiKey) {
       }
     });
 
+    // add the search pin
     map.addLayer({
       id: 'point',
       source: 'single-point',

--- a/public/src/map.js
+++ b/public/src/map.js
@@ -46,12 +46,12 @@ export async function initMap(apiKey) {
   
     // add polygon
     map.addLayer({
-      'id': 'polygons',
+      'id': 'polygons-fill',
       'type': 'fill',
       'source': 'polygons',
       'layout': {},
       'paint': {
-        'fill-color': '#088',
+        'fill-color': '#7f5a83',
         'fill-opacity': 0.3,
       }
     });
@@ -60,12 +60,12 @@ export async function initMap(apiKey) {
     // Bacause of the way the package is setup, 
     // // you cannot directly add an outline with width > 1
     map.addLayer({
-      'id': 'polygons',
+      'id': 'polygons-outline',
       'type': 'line',
       'source': 'polygons',
       'layout': {},
       'paint': {
-        'line-color': '#483D8B',
+        'line-color': '#0d324d',
         'line-width': 4
       }
     });

--- a/public/src/map.js
+++ b/public/src/map.js
@@ -37,20 +37,35 @@ export async function initMap(apiKey) {
   // add a source layer and default styling for a single point
   map.on('load', function () {
     
+    // polygon GeoJson
     map.addSource('polygons', {
       type: 'geojson',
-      data: 'https://opendata.arcgis.com/datasets/937997590f724a398ccc0100dbd9feee_0.geojson'
+      data: '..public/assets/constituencies.geojson'
     })
   
+    // add polygon
     map.addLayer({
       'id': 'polygons',
       'type': 'fill',
       'source': 'polygons',
       'layout': {},
       'paint': {
-      'fill-color': '#088',
-      'fill-opacity': 0.8,
-      'fill-outline-color': 'rgba(0,0,0,0)'
+        'fill-color': '#088',
+        'fill-opacity': 0.3,
+      }
+    });
+
+    // add polygon outline
+    // Bacause of the way the package is setup, 
+    // // you cannot directly add an outline with width > 1
+    map.addLayer({
+      'id': 'polygons',
+      'type': 'line',
+      'source': 'polygons',
+      'layout': {},
+      'paint': {
+        'line-color': '#483D8B',
+        'line-width': 4
       }
     });
     

